src/ert/config/queue_config.py

```diff
@@ -180,42 +268,82 @@ def from_dict(cls, config_dict: ConfigDict) -> QueueConfig:
-                raise ConfigValidationError(
-                    f"Invalid QUEUE_OPTION for {queue_system.name}: '{option_name}'. "
-                    f"Valid choices are {sorted(VALID_QUEUE_OPTIONS[queue_system])}."
-                )
 
-            value = values[0] if values else ""
-            if (
-                option_name in queue_options[queue_system]
-                and queue_options[queue_system][option_name] != value
-            ):
-                logger.info(
-                    f"Overwriting QUEUE_OPTION {selected_queue_system} {option_name}:"
-                    f" \n Old value: {queue_options[queue_system][option_name]} \n New value: {value}"
-                )
-            queue_options[queue_system][option_name] = value
+        _raw_queue_options = config_dict.get("QUEUE_OPTION", [])
+        _grouped_queue_options = _group_queue_options_by_queue_system(
+            _raw_queue_options
+        )
+
+        _log_duplicated_queue_options(_raw_queue_options)
+        _raise_for_defaulted_invalid_options(_raw_queue_options)
 
-        queue_options[selected_queue_system] = _add_generic_queue_options(
-            config_dict, queue_options[selected_queue_system]
+        _all_validated_queue_options = {
+            selected_queue_system: QueueOptions.create_queue_options(
+                selected_queue_system,
+                _grouped_queue_options[selected_queue_system],
+                True,
+            )
+        }
+        _all_validated_queue_options.update(
+            {
+                _queue_system: QueueOptions.create_queue_options(
+                    _queue_system,
+                    _grouped_queue_options[_queue_system],
+                    _queue_system == selected_queue_system,
+                )
+                for _queue_system in QueueSystem
+                if _queue_system != selected_queue_system
+            }
         )
 
-        if "PROJECT_CODE" not in queue_options[selected_queue_system]:
+        queue_options = _all_validated_queue_options[selected_queue_system]
+        queue_options.add_global_queue_options(config_dict)
+
+        if queue_options.project_code is None:
             tags = {
                 fm_name.lower()
                 for fm_name, *_ in config_dict.get(ConfigKeys.FORWARD_MODEL, [])
                 if fm_name in ["RMS", "FLOW", "ECLIPSE100", "ECLIPSE300"]
             }
             if tags:
-                queue_options[selected_queue_system]["PROJECT_CODE"] = "+".join(tags)
-        _check_queue_option_settings(
-            selected_queue_system, queue_options, config_dict, realization_memory
-        )
+                queue_options.project_code = "+".join(tags)
+
+        if selected_queue_system == QueueSystem.TORQUE:
+            _check_num_cpu_requirement(
+                config_dict.get("NUM_CPU", 1), queue_options, _raw_queue_options
+            )
+
+        for _queue_vals in _all_validated_queue_options.values():
```

Comment:
All these ifs can be solved by one `for loop` I guess :smile:

Reply:
from line 316 and onwards? Not so trivial I think?

Reply:
Yes, it is possible but not really pretty. Let's keep it.